### DISCUSSION TITLE
[Offline Mode] Load offline Package locally.

### DISCUSSION THREAD

**Discussion by huylocit14054** [maplibre/maplibre-react-native#537] [Q&A]
Hi, I am trying your offline manager. I could download packs and get the downloaded packs. However, I did not see any document on how to use these packs. It would be helpful if we could have an example.

**Comment by KiwiKilian** [comment]
Try to elaborate a bit more, what your want to do.

**Reply by huylocit14054** [reply]
I have a Raster Tiler that I want to download and use offline. I used the offline manager to download the Raster Tiler. However, after that, I do not know how to use the downloaded package. I can only get the package by name, and there is no option to put the package in the MapView or RasterLayer component.

**Reply by KiwiKilian** [reply]
There is not much to do with. You can simply add/remove those packs. They will be used by MapViews, which use this style, so if the pack is fully downloaded, you can go offline on your device and still see the downloaded tiles.

What else would you want to do with them?

**Reply by huylocit14054** [reply]
In that case, that was what I saw when I went offline. However, there is an edge case on this:

I have two screens, Map Screen and Map Offline Management screen. 
- The Map Screen is where I show the map and the raster tiler. 
- The Map Offline Management screen is where I have a download button. This button downloads the raster tiler from the tiles.json. I do not download the styles.json package only the raster tiler. 

**Case: Do not load the map first and only download the raster tiler package** 
_Steps:_
- Go directly to the Map Offline Management screen after installing the app. 
- Click on the download package button. The package was downloaded successfully. 
- Turn off the network and go to the Map Screen. So this is the first time the map loads, and the map shows nothing. 

_Expected_
I expect the raster tiler to load.

**Reply by huylocit14054** [reply]
https://github.com/user-attachments/assets/0b11fef0-c6c0-4d2d-82d0-62223e436c30
